### DISCUSSION TITLE
Need to insert products into saleor database from csv files

### DISCUSSION THREAD

**Discussion by Teja887** [saleor/saleor#14848] [Q&A]
Hi , 

I have a requirement to insert 6000 products into saleor database , i noticed all the database tables which will have an entry when we insert a product and entered data into those 10 tables but i am not getting to see the products in the dashboard. 
The Tables in which i entered the data are 

Product_product
product_productvariant
product_productmedia
product_variantmedia
product_productchannellisting
product_productvariantchannellisting
product_collectionproduct
attribute_assignedproductattributevalue
attribute_assignedproductattribute
warehouse_stock

The data is successfully being inserted into the tables so no foreign key issues there, In graphql api what are the data validations which are being performed when sending data to admin dashboard or storefront api. also , where are the queries location when graphql api for fetching products is called 

![image](https://github.com/saleor/saleor/assets/6998112/afa38112-b205-4028-ab8c-e0fb7e40e628)

Any help is appreciated 

Regards,
Tejeswar reddy

**Comment by timuric** [comment]
Hey @Teja887, we recommend using GraphQL to import products https://docs.saleor.io/docs/3.x/api-reference/products/inputs/product-bulk-create-input

**Comment by Teja887** [comment]
hi timuric, 
Any example of programmatically using bulk product insert !! it helps a lot.
